django_irods/views.py

```diff
@@ -187,17 +187,15 @@ def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=T
                         'download_path': '/django_irods/rest_download/' + output_path}),
                     content_type="application/json")
             else:
-                # return status to the UI
-                request.session['task_id'] = task.task_id
-                # TODO: this is mistaken for a bag download in the UI!
-                # TODO: multiple asynchronous downloads don't stack!
-                request.session['download_path'] = '/django_irods/download/' + output_path
-                # redirect to resource landing page, which interprets session variables.
-                return HttpResponseRedirect(res.get_absolute_url())
+                # return status to the task notification App AJAX call
+                task_dict = get_or_create_task_notification(task_id, name='zip download', payload=download_path,
+                                                            username=request.user.username)
+                return JsonResponse(task_dict)
```

Comment:
I am thinking in the case of anonymous user, i.e., when a user does not login and try to download a public resource, is ```request.user.username``` an empty string? Would that cause potential issues for us? What do you think? @sblack-usu

Reply:
Preparing downloads for anonymous users has not been addressed yet.  We'll need that implementation before this can be merged.